src/module.ts

```diff
@@ -246,16 +246,26 @@ class Ctrl extends PanelCtrl {
     return datasourceRequest;
   }
 
-  async export(panelId: string, target: any): Promise<void> {
-    // TODO: support org_id
-    let panelUrl = window.location.origin + window.location.pathname + `?panelId=${panelId}&fullscreen`;
+  async export(panelId: string | null, target: any): Promise<void> {
 
-    let panel = this.panels.find(el => el.id === panelId);
-    let datasourceName = panel.datasource;
+    const exportPanels = this.panels.filter(panel =>
+      panel.datasource !== undefined &&
+      panel.type !== this.selfType &&
+      (panelId === null || panel.id === panelId)
+    );
 
-    let datasourceRequest;
+    let datasources = {};
     try {
-      datasourceRequest = await this._getDatasourceRequest(datasourceName);
+      datasources = await this.backendSrv.get(`/api/datasources`) ;
+
+      const exportTable = _.keyBy(exportPanels, 'datasource');
+      datasources = (datasources as {name: string}[])
+        .reduce((result, item) => {
+        if (exportTable[item.name]) {
```

Comment:
```suggestion
        if(exportTable[item.name]) {
```

also, it's inside the reduce callback so I guess you'd want to move it 2 spaces righter